spring-integration-stomp/src/main/java/org/springframework/integration/stomp/outbound/StompMessageHandler.java

```diff
@@ -104,9 +106,11 @@ protected void onInit() throws Exception {
 
 	@Override
 	protected void handleMessageInternal(final Message<?> message) throws Exception {
-		if (!this.isRunning()) {
+		if (!this.isRunning() || !this.stompSessionManager.isConnected()) {
+			this.running = false;
+			this.stompSession = null;
 			throw new MessageDeliveryException(message, "The StompMessageHandler [" + getComponentName() +
-					"] hasn't been connected to StompSession. Check the state of [" + this.stompSessionManager + "]");
+					"] isn't connected to StompSession. Check the state of [" + this.stompSessionManager + "]");
 		}
```

Comment:
I think connection (re)establishment needs to be on-demand for the outbound adapter, rather than relying on some background recovery task - to be consistent with everything else - `rabbit`, `mqtt`, `tcp`, `ftp`, etc, etc.

We should even support lazy connections - not connect until a request is received - rather than when the adapter is started.

This is the default for most (all?) adapters - we added eager connection to amqp at the request of a user so that bad configuration could be detected earlier.